target_postgres/tests/test_standard_target.py

```diff
@@ -397,3 +421,125 @@ def test_activate_version_deletes_data_properly(postgres_config):
+
+
+def test_postgres_ssl_no_config(postgres_config_no_ssl):
+    """Test that connection will fail when no SSL configuration options are provided.
+
+    postgres_config_no_ssl has no configuration options for SSL, but port 5432 is a
+    database that requires SSL. An error is expected because connecting to this database
+    without SSL enabled shouldn't be possible.
+    """
+
+    tap = SampleTapCountries(config={}, state=None)
+
+    postgres_config_modified = copy.deepcopy(postgres_config_no_ssl)
+    postgres_config_modified["port"] = 5432
+
+    with pytest.raises(sqlalchemy.exc.OperationalError):
+        target = TargetPostgres(config=postgres_config_modified)
+        sync_end_to_end(tap, target)
+
+
+def test_postgres_ssl_no_pkey(postgres_config):
+    """Test that connection will fail when no private key is provided."""
+
+    postgres_config_modified = copy.deepcopy(postgres_config)
+    postgres_config_modified["ssl_private_key"] = None
+
+    # This is an AssertionError because checking that a private key exists is asserted
+    # for when ssl_client_certificate_enable is on.
+    with pytest.raises(AssertionError):
+        TargetPostgres(config=postgres_config_modified)
+
+
+def test_postgres_ssl_public_pkey(postgres_config):
+    """Test that connection will fail when private key access is not restricted."""
+
+    tap = SampleTapCountries(config={}, state=None)
+
+    postgres_config_modified = copy.deepcopy(postgres_config)
+    postgres_config_modified["ssl_private_key"] = "./ssl/public_pkey.key"
+
+    # If the private key exists but access is too public, the target won't fail until
+    # the it attempts to establish a connection to the database.
+    with pytest.raises(sqlalchemy.exc.OperationalError):
+        target = TargetPostgres(config=postgres_config_modified)
+        sync_end_to_end(tap, target)
+
+
+def test_postgres_ssl_no_client_cert(postgres_config):
+    """Test that connection will fail when client certificate is not provided."""
+    postgres_config_modified = copy.deepcopy(postgres_config)
+    postgres_config_modified["ssl_certificate"] = None
+
+    # This is an AssertionError because checking that a certificate exists is asserted
+    # for when ssl_client_certificate_enable is on.
+    with pytest.raises(AssertionError):
+        TargetPostgres(config=postgres_config_modified)
+
+
+def test_postgres_ssl_invalid_cn(postgres_config):
+    """Test that connection will fail due to non-matching common names.
+
+    The server is configured with certificates that state it is hosted at "localhost",
+    which won't match the loopback address "127.0.0.1". Because verify-full (the
+    default) requires them to match, an error is expected.
+    """
+    tap = SampleTapCountries(config={}, state=None)
+
+    postgres_config_modified = copy.deepcopy(postgres_config)
+    postgres_config_modified["host"] = "127.0.0.1"
+
+    with pytest.raises(sqlalchemy.exc.OperationalError):
```

Comment:
In this case I'm not too worried about it, but an FYI for later is normally when you catch an Exception you want to be specific about what kind of exception it is. In this case `OperationalError` isn't very specific so you'd have to check that the message matches the type of Exception you're expecting. I think this is alright to leave as is though!

Reply:
Just for my future reference—isn't checking the error message of the Exception a bit fragile? That seems like something that could easily change.

Reply:
Right it's better to have a specific exception for what you're trying to catch / verify. If you don't have that (as is the case here I believe) you have to use the message which is fragile, but at least you know you're catching the right exception as `OpertationalError` gets thrown for a ton of reasons